bridges/snowbridge/primitives/router/src/outbound/mod.rs

```diff
@@ -202,10 +202,13 @@ impl<'a, Call> XcmConverter<'a, Call> {
 		}
 
 		// Get the fee asset item from BuyExecution or continue parsing.
-		let fee_asset = match_expression!(self.peek(), Ok(BuyExecution { fees, .. }), fees);
-		if fee_asset.is_some() {
-			let _ = self.next();
-		}
+		let fee_asset = match_expression!(self.next()?, BuyExecution { fees, .. }, fees)
+			.ok_or(FeeAssetExpected)?;
+		ensure!(fee_asset.clone().id == AssetId::from(Location::parent()), FeeAssetInvalid);
+		let _fee_amount = match fee_asset.clone().fun {
+			Fungible(fee_amount) => Ok(fee_amount),
+			_ => Err(FeeAssetInvalid),
+		}?;
```

Comment:
Ensure the fee asset is `DOT`, question is do we still need to add some extra checks here to ensure the `fee_amount` is enough?